wezterm-gui/src/overlay/copy.rs

```diff
@@ -1507,6 +1515,23 @@ impl Pane for CopyOverlay {
     }
 }
 
+pub struct SearchOverlayPatternWriter {
+    render: Arc<Mutex<CopyRenderable>>,
+}
+
+impl std::io::Write for SearchOverlayPatternWriter {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let mut render = self.render.lock();
+        render.pattern.push_str(std::str::from_utf8(buf).unwrap());
```

Comment:
Even though in this particular use case, the incoming bytes are produced from writing a `String`, which is guaranteed to be UTF-8, the `Write` trait could be triggered from other situations and arbitrary byte sequences are potentially possible.

The `unwrap` here is undesirable, as it can cause the program to terminate with no good opportunity to catch and handle the error.

Since `write` can return `Result`, we can propage the error instead, something like this:

```suggestion
        let s = std::str::from_utf8(buf).map_err(|err| std::io::Error::new(std::io::ErrorKind::Other, format!("invalid UTF-8: {err:#}")))?;
        render.pattern.push_str(s);
```